### PR TITLE
chore: update contributing guide with merit badger info

### PR DESCRIPTION
**Pull request by kaizencc**
This PR piggybacks on #21898.

### CODE REVIEW

**Review by corymhall — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1116,6 +1117,24 @@ $ node --inspect-brk /path/to/aws-cdk/node_modules/.bin/jest -i -t 'TESTNAME'
 
 3. On the `Run` pane of VSCode, select the run configuration **Attach to NodeJS** and click the button.
 
+## Badges (Pilot Program)
+
+> CDK Merit Badges is a Pilot Program. The badges you get are experimental and may change.
+
+CDK Merit Badges is a program aimed at enhancing the CDK contributor experience. When you
+submit new pull requests to the CDK repository, you will receive a merit badge that reflects
```
[the remark this answers is not in the document]
Comment (kaizencc) [reply CONTRIBUTING.md:1125]: The `beginning-contributor` badge is for users who have 0-2 contributions. So, everyone.

[the remark this answers is not in the document]
Comment (Naumel) [reply CONTRIBUTING.md:1125]: So I'm to read it as: you have to first submit a new PR in order for all others to be counted and for the contributor to get a badge.

But what happens in 2 months?
The statement becomes outdated?

[the remark this answers is not in the document]
Comment (kaizencc) [reply CONTRIBUTING.md:1125]: Well the badges don't actually live on outside of the PR. We don't keep track of a user -> badge. Instead, we re-calculate what a user's badge would be on every opened PR.

[the remark this answers is not in the document]
Comment (Naumel) [reply CONTRIBUTING.md:1125]: You're right, I kept thinking GitHub `achievements` for some reason.
LGTM

**Review by Naumel — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaizencc) [reply CONTRIBUTING.md:1125]: I wish we could tap into github achievements! that would be so cool.

**Pull request merged**